engine/crates/tracing/src/span/gql.rs

```diff
@@ -0,0 +1,57 @@
+use tracing::{info_span, Span};
+
+use crate::span::{GqlRecorderSpanExt, GqlResponseAttributes};
+
+pub const SPAN_NAME: &str = "graphql";
+
+/// A span for a graphql request
+#[derive(Default)]
+pub struct GqlRequestSpan<'a> {
+    /// True if the response contains errors
+    has_errors: Option<bool>,
+    /// The operation name from the graphql query
+    operation_name: Option<&'a str>,
+    /// query|mutation|subscription
+    operation_type: Option<&'a str>,
+    /// The GraphQL query
+    document: Option<&'a str>,
+}
+
+impl<'a> GqlRequestSpan<'a> {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn with_document(mut self, document: impl Into<Option<&'a str>>) -> Self {
+        self.document = document.into();
+        self
+    }
+
+    pub fn with_operation_name(mut self, operation_name: impl Into<Option<&'a str>>) -> Self {
```

Comment:
operation name & operation type would always be added in the same step, we need to parse the query to be sure to have them